### PR TITLE
rebase: refactor test your assistant docs

### PR DESCRIPTION
**Pull request by erohmensing**
**Proposed changes**:
- New PR for https://github.com/RasaHQ/rasa/pull/6778. The last one was messed up because of a wrong rebase I (@dakshvar22) did.

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/master/changelog) for instructions)
- [ ] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

**Comment by dakshvar22** [comment]
Adding @alopez in the loop on the new PR.

File: docs/docs/testing-your-assistant.mdx [hunk 1]
```diff
@@ -3,22 +3,22 @@ id: testing-your-assistant
 sidebar_label: Testing Your Assistant
 title: Testing Your Assistant
 abstract: Rasa Open Source lets you test dialogues end-to-end by running through
-  test stories. You can also test the dialogue management and the message processing 
-  (NLU) separately.
+  test conversations. In addition to full test conversations, you can
```
Comment (erohmensing) [inline docs/docs/testing-your-assistant.mdx:6]: I believe we settled on `test stories` as the preferred term. See thread: https://rasa-hq.slack.com/archives/C016FLYV4QN/p1600259900015600

File: docs/docs/testing-your-assistant.mdx [hunk 2]
```diff
@@ -3,22 +3,22 @@ id: testing-your-assistant
...
 ---
 
-## End-to-end Testing
+## Writing Test Conversations
 
-Testing your assistant requires you to write test stories, which include
-the user messages and the conversation history. The format is a modified version of [the one
-used to specify stories in your training data](./stories.mdx#format).
+To test your assistant completely, the best approach is to write test conversations. These are like
+stories in your training data, but with the actual user utterance included as well.
```
Comment (erohmensing) [inline docs/docs/testing-your-assistant.mdx:14]:
```suggestion
To test your assistant completely, the best approach is to write test conversations. Test stories look like the
stories in your training data, but include the actual user utterance as well.
```

Comment (erohmensing) [reply docs/docs/testing-your-assistant.mdx:14]: obviously test stories everywhere, but

File: docs/docs/testing-your-assistant.mdx [hunk 3]
```diff
@@ -117,113 +117,142 @@ Here are some examples:
...
 You can test your assistant against them by running:
 
 ```bash
 rasa test
 ```
 
-The command will always load all stories from any story files with filenames
-starting with `test_`, e.g. `test_stories.yml`. Your story test
-file names should always start with `test_` for this detection to work.
+See the [CLI documentation on rasa test command](./command-line-interface.mdx#rasa-test) for
```
Comment (erohmensing) [inline docs/docs/testing-your-assistant.mdx:127]:
```suggestion
See the [CLI documentation on `rasa test`](./command-line-interface.mdx#rasa-test) for
```

File: docs/docs/testing-your-assistant.mdx [hunk 4]
```diff
@@ -117,113 +117,142 @@ Here are some examples:
...
 ```
 
-If you've done this, you can see how well your NLU model predicts the
-test cases:
+Next, you can see how well your trained NLU model predicts the
+test cases using:
 
-```bash
-rasa test nlu --nlu train_test_split/test_data.yml
+```bash {2}
```
Comment (erohmensing) [inline docs/docs/testing-your-assistant.mdx:155]: Why are we highlighting here?

File: docs/docs/testing-your-assistant.mdx [hunk 5]
```diff
@@ -117,113 +117,142 @@ Here are some examples:
...
 
-By passing multiple pipeline configurations (or a folder containing them) to the CLI, Rasa will run
-a comparison between the pipelines:
+To get the most out of your training data, you should train and evaluate your model on different pipelines
+and different amounts of training data.
 
-```bash
-rasa test nlu --config config_1.yml config_2.yml
-  --nlu data/nlu.yml --runs 3 --percentages 0 25 50 70 90
+To do so, use:
```
Comment (erohmensing) [inline docs/docs/testing-your-assistant.mdx:183]:
```suggestion
To do so, pass multiple configuration files to the `rasa test` command:
```

File: docs/docs/testing-your-assistant.mdx [hunk 6]
```diff
@@ -117,113 +117,142 @@ Here are some examples:
...
+three times for each configuration specified.
+A graph with the mean and standard deviations of
+[f1-scores](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.f1_score.html)
+across all runs is plotted.
 
 The f1-score graph, along with all train/test sets, the trained models, classification and error reports,
 will be saved into a folder called `nlu_comparison_results`.
 
-### Intent Classification
+Inspecting the f1-score graph can help in understanding if you have enough data for your NLU model.
```
Comment (erohmensing) [inline docs/docs/testing-your-assistant.mdx:207]:
```suggestion
Inspecting the f1-score graph can help you understand if you have enough data for your NLU model.
```

Comment (erohmensing) [reply docs/docs/testing-your-assistant.mdx:207]: Should this section not go into "Interpreting the output"?

File: docs/docs/testing-your-assistant.mdx [hunk 7]
```diff
@@ -117,113 +117,142 @@ Here are some examples:
...
 
-The report logs precision, recall and f1 measure for each intent and entity,
+The report logs [precision](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.precision_score.html),
+[recall](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.recall_score.html) and
+[f1-score](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.f1_score.html) for each intent,
 as well as providing an overall average. You can save these reports as JSON files using the `--report` argument.
 
-The confusion matrix shows you which intents are mistaken for others;
-any samples which have been incorrectly predicted are logged and saved to a file called `errors.json` for easier debugging.
+The confusion matrix shows which intents are mistaken for others.
```
Comment (erohmensing) [inline docs/docs/testing-your-assistant.mdx:230]: let's give filenames for the confmat and the histogram

File: docs/docs/testing-your-assistant.mdx [hunk 8]
```diff
@@ -117,113 +117,142 @@ Here are some examples:
...
+Any samples which have been incorrectly predicted are logged and saved to a file called `errors.json` for easier debugging.
 
-The histogram that the script produces allows you to visualise the confidence distribution for all predictions, with the volume of correct and incorrect predictions being displayed by blue and red bars respectively. Improving the quality of your training data will move the blue histogram bars to the right and the red histogram bars to the left of the plot.
+The histogram allows you to visualize the confidence for all predictions,
+with the correct and incorrect predictions being displayed by blue and red bars respectively.
+Improving the quality of your training data will move the blue histogram bars to the right and the
+red histogram bars to the left of the plot.
```
Comment (erohmensing) [inline docs/docs/testing-your-assistant.mdx:236]: and will hopefully lead to less red, right? 🙂

File: docs/docs/testing-your-assistant.mdx [hunk 9]
```diff
@@ -117,113 +117,142 @@ Here are some examples:
...
 
-### Entity Extraction
+#### Entity Extraction
 
-Only trainable entity extractors, such as the `DIETCLassifier` and `CRFEntityExtractor` will be
-evaluated by the `rasa test` script. If you pretrained extractors like the `DucklingHTTPExtractor`
-Rasa Open Source will not include these in the evaluation.
-
-`rasa test` reports recall, precision, and f1 measure for each entity type that
+`rasa test` also reports recall, precision, and f1-score for each entity type that
```
Comment (erohmensing) [inline docs/docs/testing-your-assistant.mdx:250]:
```suggestion
`rasa test` reports recall, precision, and f1-score for each entity type that
```

File: docs/docs/testing-your-assistant.mdx [hunk 10]
```diff
@@ -117,113 +117,142 @@ Here are some examples:
...
+with the correct and incorrect predictions being displayed by blue and red bars respectively.
+Improving the quality of your training data will move the blue histogram bars to the right and the
+red histogram bars to the left of the plot.
 
-### Response Selection
+#### Response Selectors
 
-The evaluation script will produce a combined report for all response selector models in your pipeline.
+In addition to intent classifiers, `rasa test` also evaluates response selectors, producing a
+report, confusion matrix, confidence histogram and errors.
```
Comment (erohmensing) [inline docs/docs/testing-your-assistant.mdx:241]:
```suggestion
`rasa test` evaluates response selectors in the same way that it evaluates intent classifiers, producing a
report, confusion matrix, confidence histogram and errors.
```

File: docs/docs/testing-your-assistant.mdx [hunk 11]
```diff
@@ -117,113 +117,142 @@ Here are some examples:
...
 
-Only trainable entity extractors, such as the `DIETCLassifier` and `CRFEntityExtractor` will be
-evaluated by the `rasa test` script. If you pretrained extractors like the `DucklingHTTPExtractor`
-Rasa Open Source will not include these in the evaluation.
-
-`rasa test` reports recall, precision, and f1 measure for each entity type that
+`rasa test` also reports recall, precision, and f1-score for each entity type that
 your trainable entity extractors are trained to recognize.
 
+Only trainable entity extractors, such as the `DIETClassifier` and `CRFEntityExtractor` will be
```
Comment (erohmensing) [inline docs/docs/testing-your-assistant.mdx:253]:
```suggestion
Only trainable entity extractors, such as the `DIETClassifier` and `CRFEntityExtractor` are
```

File: docs/docs/testing-your-assistant.mdx [hunk 12]
```diff
@@ -300,9 +328,11 @@ rasa train core -c config_1.yml config_2.yml \
   --out comparison_models --runs 3 --percentages 0 5 25 50 70 95
 ```
 
+Similar to how the NLU model was evaluated the above command trains the dialogue model
```
Comment (erohmensing) [inline docs/docs/testing-your-assistant.mdx:331]: add a link to that section, comma after `evaluated`

File: docs/docs/testing-your-assistant.mdx [hunk 13]
```diff
@@ -312,14 +342,18 @@ rasa test core -m comparison_models --stories stories_folder
...
 To compare single policies, create config files containing only one policy each.
 
 :::note
 This training process can take a long time, so we'd suggest letting it run
 somewhere in the background where it can't be interrupted.
 
 :::
+
+If you have any questions or problems, please share them with us in the dedicated
+[testing section on our forum](https://forum.rasa.com/tags/testing) !
```
Comment (erohmensing) [inline docs/docs/testing-your-assistant.mdx:359]:
```suggestion
[testing section on our forum](https://forum.rasa.com/tags/testing)!
```

**Review by erohmensing — COMMENTED** [review]
@dakshvar22 a review for you 🙂

**Comment by erohmensing** [comment]
omg i use `test` to rebase branches too often 😅

**Comment by dakshvar22** [comment]
@erohmensing Thanks for the comment. I had almost forgotten I had to come back to your review on this PR 🙈 . On it now

**Comment by erohmensing** [comment]
No worries, the comment was totally accidental as i accidentally force-pushed Sam's rebase to this branch lol

File: docs/docs/testing-your-assistant.mdx (hunk 4 above)
Comment (dakshvar22) [reply docs/docs/testing-your-assistant.mdx:155]: To show how to define the file containing the NLU test cases.

File: docs/docs/testing-your-assistant.mdx (hunk 6 above)
Comment (dakshvar22) [reply docs/docs/testing-your-assistant.mdx:207]: This F1 score graph is particularly for the scenario when you are comparing multiple pipelines which is this particular section. `Interpreting the Output` section describes the output of `rasa test` command in the default mode which is for just one config.

File: docs/docs/testing-your-assistant.mdx (hunk 1 above)
Comment (dakshvar22) [reply docs/docs/testing-your-assistant.mdx]: Changed it to test stories.

File: docs/docs/testing-your-assistant.mdx (hunk 8 above)
Comment (dakshvar22) [reply docs/docs/testing-your-assistant.mdx]: Added that bit.

**Comment by dakshvar22** [comment]
@erohmensing Ready for another review 🙏

File: docs/docs/testing-your-assistant.mdx (hunk 6 above)
Comment (erohmensing) [reply docs/docs/testing-your-assistant.mdx]: Ah, that makes sense 👌

File: docs/docs/testing-your-assistant.mdx [hunk 14]
```diff
@@ -117,113 +117,144 @@ Here are some examples:
...
+Improving the quality of your training data will move the blue histogram bars to the right and the
+red histogram bars to the left of the plot. It should also help in reducing the number red histogram bars itself.
 
-### Response Selection
+#### Response Selectors
 
-The evaluation script will produce a combined report for all response selector models in your pipeline.
+`rasa test` evaluates response selectors in the same way that it evaluates intent classifiers, producing a
+report(`response_selection_report.json`, confusion matrix(`response_selection_confusion_matrix.png`),
+confidence histogram(`response_selection_histogram.png`) and errors(`response_selection_errors.json`).
```
Comment (erohmensing) [inline docs/docs/testing-your-assistant.mdx:243]:
```suggestion
report (`response_selection_report.json`, confusion matrix (`response_selection_confusion_matrix.png`),
confidence histogram (`response_selection_histogram.png`) and errors(`response_selection_errors.json`).
```

**Review by erohmensing — COMMENTED** [review]
Looking good 🥇

**Comment by dakshvar22** [comment]
@erohmensing Addressed all comments. Thanks for bringing up the smaller details I missed.

**Review by dakshvar22 — APPROVED** [review]
Thanks for the review @erohmensing 😄

**Pull request merged**